### DISCUSSION TITLE
How to use Logmon when running as ContainerUser

### DISCUSSION THREAD

**Discussion by rigdal** [microsoft/windows-container-tools#142] [Q&A] [answered]
Running apps in containers as a non-root user is standard with containers nowadays. 

I am new to Windows containers, but it appears that I should leverage `USER ContainerUser` in my Dockerfile for running as non-root. 

However, I cannot seem to start Logmon or maybe there is a better spot to have my USER layer in my Dockerfile. Here is the output I get upon startup.

_npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] ERROR: Failed to start ETW trace. Error: 5
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] ERROR: Failed to start ETW trace session. Error: 5
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] ERROR: Failed to start ETW monitor. Error: 5
npc-oes-webapp  | 
npc-oes-webapp  | ERROR: Failed to stop or query status of service 'w3svc' error [80070005]
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] ERROR: Failed to open log directory handle. Directory: \\?\c:\inetpub\logs Error=5
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] ERROR: Failed to start log file monitor. Log files in a directory \\?\c:\inetpub\logs will not be monitored. Error: 5
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] INFO: Entrypoint processs exit code: -2147024891
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] WARNING: The given session is not running. Error: 4201
npc-oes-webapp  | [2023-05-15T16:48:49.000Z][LOGMONITOR] ERROR: Failed to wait for log file monitor to stop. Log directory: \\?\c:\inetpub\logs Error: 2147942406_

**Comment by profnandaa** [comment, marked as answer]
Opened an issue here #143; we're investigating.
